app/assets/javascripts/discourse/lib/text-direction.js.es6

```diff
@@ -0,0 +1,28 @@
+const ltrChars = 'A-Za-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02B8\u0300-\u0590\u0800-\u1FFF\u2C00-\uFB1C\uFDFE-\uFE6F\uFEFD-\uFFFF';
+const rtlChars = '\u0591-\u07FF\uFB1D-\uFDFD\uFE70-\uFEFC';
+
+export function isRTL(text) {
+  const rtlDirCheck = new RegExp('^[^'+ltrChars+']*['+rtlChars+']');
```

Comment:
Rather than parsing this every time it's called, perhaps you should move `rtlDirCheck` outside this function as a constant? Then it would only be parsed once.  (Same with `ltrDirCheck` below)

The only gotcha is you have to reset `lastIndex` before you use it., but it should be more efficient.